packages/block-editor/src/components/page-template-picker/use-page-template-picker-visible.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+
+const __experimentalUsePageTemplatePickerVisible = () => {
+	return useSelect( ( select ) => {
+		const {
+			getCurrentPostType,
+		} = select( 'core/editor' );
+
+		const {
+			getBlockCount,
+			getSettings,
+		} = select( 'core/block-editor' );
+
+		const isPageTemplatesEnabled = getSettings().__experimentalEnablePageTemplates;
+		const isEmpty = getBlockCount() === 0;
```

Comment:
Wdyt about deciding `isEmpty` by checking the html content instead of block count? That way we'll show the buttons until user starts to type and once the user deletes everything the buttons would be visible again. I found it a bit hard to make buttons visible again by undo button.

Reply:
I don't like looking at the HTML (which means having to serialize), but I think we can achieve the same effect with `isUnmodifiedDefaultBlock` if there is only one block, I'll give it a try